pkg/reflectutil/reflectutil.go

```diff
@@ -0,0 +1,104 @@
+package reflectutil
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+
+	"github.com/huandu/go-clone"
+)
+
+// NonAppendableSliceTypes are non-appendable slice types, such as net.IP .
+var NonAppendableSliceTypes = map[reflect.Type]struct{}{
+	reflect.TypeOf(net.IP{}): {}, // []byte
+}
+
+// MergeMany merges vv and returns the new value.
+// MergeMany does not alter vv.
+//
+// Maps are merged in "vv[0], vv[1], .., vv[N-1]" order.
+// Slices are appended in the "vv[N-1], .., vv[1], vv[0]" order.
+func MergeMany(vv ...interface{}) (interface{}, error) {
+	if l := len(vv); l < 2 {
+		return nil, fmt.Errorf("expected len(vv) >= 2, got %d", l)
+	}
+	x := vv[0]
+	for _, v := range vv[1:] {
+		var err error
+		x, err = Merge(x, v)
+		if err != nil {
+			return x, err
```

Comment:
Don't return incompletely merged result in case of error:

```suggestion
			return nil, err
```